app/src/main/java/com/example/cinemapp/ui/authentication/AuthenticationActivity.kt

```diff
@@ -63,7 +63,9 @@ class AuthenticationActivity : AppCompatActivity() {
         lifecycleScope.launch {
             viewModel.signInAttempt.collect { success ->
                 if (success) {
-                    finishThenStart(this@AuthenticationActivity, MainActivity::class.java)
+                    if (isTaskRoot)
```

Comment:
Try to group lines of this `if` statement into a private method and give it a meaningful name